R/settings.R

```diff
@@ -49,18 +38,41 @@ read_settings <- function(filename) {
   }
 }
 
+read_config_file <- function(config_file) {
+  if (is.null(config_file)) {
+    return(NULL)
+  }
+
+  config <- new.env()
+  if (endsWith(config_file, ".R")) {
+    load_config <- function(file) sys_source(file, config)
```

Comment:
Does the code need to import `lintr` like this or can it use, e.g., `linters <- all_linters()`?

Reply:
Not stress-tested, but I did copy over `modify_defaults()` & that worked, so I suppose `all_linters()` will as well. Will keep in mind when writing tests.

Reply:
`linters_with_defaults()` is included in the tests, LMK if you think we need to test more thoroughly.